air/tests/test_module/instructions/fail.rs

```diff
@@ -97,6 +124,34 @@ fn fail_with_last_error_tetraplets() {
     );
 }
 
+#[test]
+fn fail_with_error_tetraplets() {
+    let local_peer_id = "local_peer_id";
+    let fallible_service_id = "service_id_1";
+    let (host_closure, tetraplet_anchor) = tetraplet_host_function(fallible_call_service(fallible_service_id));
+    let mut vm = create_avm(host_closure, local_peer_id);
+
+    let local_fn_name = "local_fn_name";
+    let script = format!(
+        r#"
+        (xor
+            (xor
+                (call "{local_peer_id}" ("{fallible_service_id}" "{local_fn_name}") [] result_1)
+                (fail :error:)
+            )
+            (call "{local_peer_id}" ("" "") [%last_error%])
```

Comment:
What do you think about assigning an `:error:` object to a scalar inside the inner xor and send the scalar as the second call argument to avoid using `%last_error%` that will become obsolete?  
```suggestion
            (call "{local_peer_id}" ("" "") [error_obj_scalar])
```